subscriber/src/daily_operations.rs

```diff
@@ -0,0 +1,250 @@
+
+        let run_result = self.run_while_it_has_gas(GAS_TO_SAVE_PROGRESS, || {
+            self.perform_one_operation::<SC>(
+                &mut progress,
+                &mut all_data,
+                &mut output_egld,
+                &mut output_esdt,
+            )
+        });
+
+        if run_result == OperationCompletionStatus::InterruptedBeforeOutOfGas {
+            self.save_progress(&progress);
+        }
+
+        *user_index = progress.additional_data_index;
+
+        let caller = self.blockchain().get_caller();
+        if output_egld > 0 {
+            self.send().direct_egld(&caller, &output_egld);
+        }
+        if !output_esdt.is_empty() {
+            self.send().direct_multi(&caller, &output_esdt);
+        }
+
+        run_result
+    }
+
+    fn perform_one_operation<SC: SubscriberContract<SubSc = Self>>(
+        &self,
+        progress: &mut OperationProgress,
+        all_data: &mut OperationData<Self::Api, SC::AdditionalDataType>,
+        output_egld: &mut BigUint,
+        output_esdt: &mut ManagedVec<EsdtTokenPayment>,
+    ) -> LoopOp {
+        if progress.additional_data_index >= all_data.additional_data_len
+            || progress.current_index > all_data.total_users
+        {
+            return STOP_OP;
+        }
+
+        let user_data = all_data.additional_data.get(all_data.user_index);
+        progress.additional_data_index += 1;
+
+        let user_id = all_data
+            .users_mapper
+            .get_by_index_at_address(&all_data.fees_contract_address, progress.current_index);
+        let opt_user_address = self
+            .user_id()
+            .get_address_at_address(&all_data.fees_contract_address, user_id);
+        let user_address = match opt_user_address {
+            Some(address) => address,
+            None => {
+                return CONTINUE_OP;
+            }
+        };
+
+        progress.current_index += 1;
+
+        let subscription_type = self
+            .subscription_type(all_data.service_id, user_id, all_data.service_index)
+            .get_from_address(&all_data.fees_contract_address);
+        let next_action_epoch =
+            self.get_next_action_epoch(user_id, all_data.service_index, subscription_type);
+        if next_action_epoch > all_data.current_epoch {
+            return CONTINUE_OP;
+        }
+
+        let subtract_result = self.call_subtract_payment(
+            all_data.fees_contract_address.clone(),
+            all_data.service_index,
+            user_id,
+        );
+        if subtract_result.is_err() {
+            return CONTINUE_OP;
+        }
+
+        let fee = unsafe { subtract_result.unwrap_unchecked() };
+        if fee.token_identifier.is_egld() {
+            *output_egld += fee.amount;
+        } else {
+            let payment = EsdtTokenPayment::new(fee.token_identifier.unwrap_esdt(), 0, fee.amount);
+            output_esdt.push(payment);
+        }
+
+        let action_results = SC::perform_action(
```

Comment:
actually we do several actions per one user per one payment. Like we claim boosted rewards for all his farms and fees collector.

Reply:
Each subscriber contract can do whatever they want in their `perform_action` implementation. If they want to perform several actions for one payment, that's fine.